src/api/index.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+import deepFreeze from "deep-freeze";
+
+// Exports for calling SourceCred code programmatically. Both the
+// structure and the contents of this API are experimental and subject
+// to change.
+import * as address from "../core/address";
+import * as discourseAddress from "../plugins/discourse/address";
+import * as discourseDeclaration from "../plugins/discourse/declaration";
+import * as githubDeclaration from "../plugins/github/declaration";
+import * as githubEdges from "../plugins/github/edges";
+import * as githubNodes from "../plugins/github/nodes";
+import * as graph from "../core/graph";
+import * as graphToMarkovChain from "../core/attribution/graphToMarkovChain";
+import * as markovChain from "../core/attribution/markovChain";
+import * as timelineCred from "../analysis/timeline/timelineCred";
+
+export default deepFreeze({
+  analysis: {
+    timeline: {
+      timelineCred,
+    },
+  },
+  core: {
+    address,
+    attribution: {
+      markovChain,
+      graphToMarkovChain,
+    },
+    graph,
+  },
+  plugins: {
+    github: {
+      declaration: githubDeclaration,
+      edges: githubEdges,
+      nodes: githubNodes,
+    },
+    discourse: {
+      address: discourseAddress,
+      declaration: discourseDeclaration,
+    },
+  },
```

Comment:
Wondering if it makes sense to let this be a mixin step.
So the default export would look like:
```js
{
  ...core,
  plugins,
}
```

Reply:
I’m not sure what you’re suggesting. What’s a “mixin step”? Why would we
want to spread `core` into the export but store `plugins` on the
attribute intact?

This whole structure is mainly intended to serve as a bridge to analysis
in Observable notebooks, and is explicitly not under any API guarantees,
so I’m not super inclined to bikeshed about the exact shape of the
export.

Reply:
Well, from the core side of things the use-case of exposing the essentials as a library is something we're going to want for the long term. Plugins on the other hand are kind of hitching a ride on this API while they're still baked into the monolith.

So to me it would make sense we declare an API for _just_ the core components so we can refine this over time to be an API we're happy with. The plugin API doesn't need that treatment for the foreseeable future.

Hence I'm suggesting we make that difference explicit with separate declarations, and letting the plugins hitch a ride at the final step. Call it mixing in, or composing. It's typically the only thing an `index.js` file does.

Reply:
I still don’t understand what change is actually being proposed. We do
need to export both the core and the plugins for now, because plugin
declarations are required to get the weights, which is required for
analysis. As written, the default export here exactly mirrors the
directory structure, which seems like a simple and uncomplicated way of
exposing our existing code without introducing additional friction or
cognitive overhead.

The one person who has used this for actual work is @decentralion, so
I’ll let them weigh in on whether they think that the plugin exports
need to be somehow separated from the rest.